src/js/components/modals/ServiceFormModal.js

```diff
@@ -307,6 +308,14 @@ class ServiceFormModal extends mixin(StoreMixin) {
     }
 
     if (this.state.jsonMode) {
+
+      // Handle service rename if this action renamed a field
+      if (this.props.isEdit && this.props.service && this.props.service.getId()) {
+        if (this.state.service.getId() !== this.props.service.getId()) {
+          this.state.service.handleRename( this.props.service.getId(), this.state.service.getId() );
```

Comment:
1. We are more than 80 characters here.
2. There are also unnecessary spaces in the function call.
3. Finally, this seems to build on a bigger problem of having multiple sources of truth. Not sure we can do much about it, though :/